pytest/tests/loadtest/locust/common/base.py

```diff
@@ -367,6 +373,9 @@ def send_tx_async(self, tx: Transaction, locust_name: str) -> dict:
         self.request_event.fire(**meta)
         return meta
 
+    # It's ok to use a slightly out-of-date block hash and this avoids one additional RPC request on
+    # every transaction.
+    @cached(cache=TTLCache(maxsize=1, ttl=1))
     def final_block_hash(self):
```

Comment:
what is this block hash used for? are we saying that using a different block for an RPC still works?

Reply:
It is used to specify the `base_block_hash` of a transaction that is used to disallow transactions that are too old to be submitted to the network: https://github.com/near/nearcore/blob/524dc6536a0c439d617e9b83d1a81bd29b9b1cea/chain/chain/src/store/mod.rs#L702

As long as `transaction_validity_period` is large enough (on mainnet it is equal to 86400 which corresponds to 1 day), using a slightly older block hash is fine.

There is some documentation about this in https://github.com/near/NEPs/blob/48c9f160b1d74f9f48d644a36ee8930c02331d5b/specs/RuntimeSpec/Scenarios/FinancialTransaction.md?plain=1#L53